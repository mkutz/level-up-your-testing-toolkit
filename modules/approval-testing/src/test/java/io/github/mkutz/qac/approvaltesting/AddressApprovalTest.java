package io.github.mkutz.qac.approvaltesting;

import static io.github.mkutz.qac.approvaltesting.AddressBuilder.anAddress;
import static io.github.mkutz.qac.approvaltesting.FakeFunctionalityKt.anOrderWasProcessed;
import static io.github.mkutz.qac.approvaltesting.FakeFunctionalityKt.callRestEndpointForBillingAddress;
import static io.github.mkutz.qac.approvaltesting.TestOrderBuilderKt.anyOrder;

import org.approvaltests.JsonApprovals;
import org.junit.jupiter.api.Test;

class AddressApprovalTest {
  @Test
  void assertionTest() {
    String orderId = "someOrderId";
    ShopOrder shopOrder =
        anyOrder(orderId)
            .billingAddress(
                anAddress()
                    .id("someBillingAddressId")
                    .firstName("Micha")
                    .lastName("Kutz")
                    .streetName("Domstr.")
                    .houseNumber("20")
                    .postalCode("50668")
                    .city("KÃ¶ln")
                    .country("Deutschland")
                    .phone("+49 221 1490")
                    .email("info@rewe-group.com")
                    .build())
            .build();

    anOrderWasProcessed(shopOrder);

    JsonApprovals.verifyJson(callRestEndpointForBillingAddress(orderId));
  }
}
