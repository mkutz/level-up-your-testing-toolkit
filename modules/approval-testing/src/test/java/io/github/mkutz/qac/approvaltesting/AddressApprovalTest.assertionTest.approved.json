{
  "id": "someBillingAddressId",
  "firstName": "Micha",
  "lastName": "Kutz",
  "streetName": "Domstr.",
  "houseNumber": "20",
  "city": "KÃ¶ln",
  "country": "Deutschland",
  "phone": "+49 221 1490",
  "latitude": "50.94603935915518",
  "longitude": "6.959302840118697",
  "email": "info@rewe-group.com",
  "postalCode": "50668",
  "status": "new_customer"
}